packages/client/src/views/CorrectionForm/utils.ts

```diff
@@ -92,6 +94,85 @@ export function isCorrection(application: IApplication) {
   return application.registrationStatus === SUBMISSION_STATUS.REGISTERED
 }
 
+export function updateApplicationRegistrationWithCorrection(
+  application: IApplication,
+  meta?: { userPrimaryOffice?: GQLLocation }
+): void {
+  const correctionValues: Record<string, any> = {}
+  const { data } = application
+
+  if (data.corrector && data.corrector.relationship) {
+    correctionValues.requester = ((
+      data.corrector.relationship as IFormSectionData
+    ).value || data.corrector.relationship) as string
+  }
+
+  if (data.reason) {
+    if (data.reason.type) {
+      correctionValues.reason = ((data.reason.type as IFormSectionData).value ||
+        data.reason.type) as string
+    }
+
+    if (data.reason.additionalComment) {
+      correctionValues.note = data.reason.additionalComment
+    }
+  }
+
+  if (data.supportingDocuments) {
+    if (
+      typeof data.supportingDocuments.supportDocumentRequiredForCorrection ===
+      'boolean'
+    ) {
+      if (data.supportingDocuments.supportDocumentRequiredForCorrection) {
+        correctionValues.hasShowedVerifiedDocument = true
+      } else {
+        correctionValues.noSupportingDocumentationRequired = true
+      }
+    }
+
+    if (data.supportingDocuments.uploadDocForLegalProof) {
+      correctionValues.data = (
+        data.supportingDocuments.uploadDocForLegalProof as IAttachmentValue
+      ).data
+    }
+  }
+
+  if (data.currectionFeesPayment) {
+    if (
+      (data.currectionFeesPayment.correctionFees as IFormSectionData)?.value &&
+      (data.currectionFeesPayment.correctionFees as IFormSectionData).value ===
+        'REQUIRED'
+    ) {
+      const { nestedFields }: { nestedFields?: IFormSectionData } = data
+        .currectionFeesPayment.correctionFees as IFormSectionData
+      correctionValues.payment = {
+        total: nestedFields?.totalFees,
+        type: 'MANUAL'
+      }
+      if (nestedFields?.proofOfPayment) {
+        correctionValues.payment.data = (
+          nestedFields?.proofOfPayment as IFileValue
+        ).data
+      }
+    }
+  }
+
+  if (meta) {
+    if (meta.userPrimaryOffice) {
+      correctionValues.location = {
+        _fhirID: meta.userPrimaryOffice._fhirID
+      }
+    }
+  }
+
+  data.registration.correction = data.registration.correction
+    ? {
+        ...(data.registration.correction as IFormSectionData),
+        ...correctionValues
+      }
+    : correctionValues
+}
+
 export function sectionHasError(
```

Comment:
I think the correction journey doesn't have to consider any error.

Reply:
Thanks @atiqzaman-dsi , but I think this PR does not add this change. But should I remove this as a part of this PR?